runtime/src/hardened_unpack.rs

```diff
@@ -27,12 +27,28 @@ pub enum UnpackError {
 
 pub type Result<T> = std::result::Result<T, UnpackError>;
 
-const MAX_SNAPSHOT_ARCHIVE_UNPACKED_SIZE: u64 = 500 * 1024 * 1024 * 1024; // 500 GiB
+// 64 TiB; some safe margin to the max 128 TiB in amd64 linux userspace VmSize
+// (ref: https://unix.stackexchange.com/a/386555/364236)
+// note that this is directly related to the mmaped data size
+// so protect against insane value
+// This is the file size including holes for sparse files
+const MAX_SNAPSHOT_ARCHIVE_UNPACKED_APPARENT_SIZE: u64 = 64 * 1024 * 1024 * 1024 * 1024;
+
+// 4 TiB;
+// This is the actually consumed disk usage for sparse files
+const MAX_SNAPSHOT_ARCHIVE_UNPACKED_ACTUAL_SIZE: u64 = 4 * 1024 * 1024 * 1024 * 1024;
```

Comment:
I don't have strong reason to add another limit. But, generally I want to add some sane value, just in case.

Even if we move to delta-based snapshot before/after secure snapshot, this code might be reused in some way in the distant future.

Reply:
Part of me wants to suggest a lower limit here, but the other half saw the offending snapshot consume 240+GB of disk during unpack.  I didn't let it finish, but wouldn't be surprised if it ticked our previous cap.

Reply:
thanks for sharing your thoughts. Even if I understand your concern, I'll just stick with this new value.

This is kind of representation of my emotion as a number. I was beaten by being too small. So, I'm making it too large. If ever should I be beaten by being that large in the future, let's reduce it. What a fantstic iteration of ~~_price_~~ limit discovery xD.

Reply:
We can binsearch our way to victory!